src/core/ext/xds/xds_http_rbac_filter.cc

```diff
@@ -412,6 +438,21 @@ Json ParseHttpRbacToJson(const envoy_extensions_filters_http_rbac_v3_RBAC* rbac,
       }
       inner_rbac_json.emplace("policies", std::move(policies_object));
     }
+    // Flatten the nested messages defined in rbac.proto
+    if (envoy_config_rbac_v3_RBAC_has_audit_logging_options(rules)) {
+      const auto* audit_logging_options =
+          envoy_config_rbac_v3_RBAC_audit_logging_options(rules);
+      inner_rbac_json.emplace(
+          "audit_condition",
+          envoy_config_rbac_v3_RBAC_AuditLoggingOptions_audit_condition(
```

Comment:
We should check that this field has a supported value.  If not, we should report an error.

Please also add a test for this case.

Reply:
Done.